src/Analyser/NodeScopeResolver.php

```diff
@@ -3625,7 +3626,16 @@ private function processAssignVar(
 			// 2. eval dimensions
 			$offsetTypes = [];
 			$offsetNativeTypes = [];
-			foreach (array_reverse($dimExprStack) as $dimExpr) {
+			$dimFetchStack = array_reverse($dimFetchStack);
+			$lastDimKey = array_key_last($dimFetchStack);
+			foreach ($dimFetchStack as $key => $dimFetch) {
+				$dimExpr = $dimFetch->dim;
+
+				// Callback was already called for last dim at the beginning of the method.
+				if ($key !== $lastDimKey) {
+					$nodeCallback($dimFetch, $enterExpressionAssign ? $scope->enterExpressionAssign($dimFetch) : $scope);
```

Comment:
I'm not sure whether the order of `$nodeCallback` calls is correct. First it's called with the outermost dim-fetch, and then the inner dim-fetches starting from the left. It would seem more natural to call the outermost dim-fetch last, but I wasn't able to make it work that way.